### PR TITLE
Slack Notifications & General Monitoring Overhaul

### PR DESCRIPTION
**Pull request by ddvdozuki**
This will overhaul and improve our monitoring capabilities in Private Cloud by:

1. Removal of cloudwatch helm chart, this is no longer the accepted best practice for getting logs and metrics into cloudwatch from EKS. We now use the ADOT (AWS distro for open telemetry) helm chart instead. It also gives us the ability to easily stream to prometheus in the future.
2. Major refactor to our EKS and app alarms. Many of the alarms we created were not done correctly in the first place or were depending on telemetry from the EKS cluster that had been disabled 🤦 
3. Addition of an alarm for DMS task failure so we'll get a notification in slack or via email. 
4. Slack & Email alarms. We now have the option to setup an email subscription and/or slack notifications for all monitoring alarms. This required the addition of our first in-terraform lambda to handle slack specifically. Additionally I had to create a new small slack app to handle the requests from the lambda. There is no code for this app, it's simply a wrapper for a webhook url.

You can safely ignore the `logical/charts/*` changes as that is just including the new ADOT helm chart and removing the old cloudwatch chart.

### CODE REVIEW

**Review by scbarber — APPROVED** [review]
LGTM!

**Review by scbarber — APPROVED** [review]

File: terraform/logical/README.md [hunk 1]
```diff
@@ -31,10 +31,10 @@ No modules.
 | Name | Type |
 |------|------|
 | [aws_autoscaling_lifecycle_hook.aws_node_termination_handler](https://registry.terraform.io/providers/hashicorp/aws/4.57.0/docs/resources/autoscaling_lifecycle_hook) | resource |
+| [helm_release.adot_exporter](https://registry.terraform.io/providers/hashicorp/helm/2.9.0/docs/resources/release) | resource |
 | [helm_release.aws_node_termination_handler](https://registry.terraform.io/providers/hashicorp/helm/2.9.0/docs/resources/release) | resource |
 | [helm_release.cert_manager](https://registry.terraform.io/providers/hashicorp/helm/2.9.0/docs/resources/release) | resource |
 | [helm_release.cluster_autoscaler](https://registry.terraform.io/providers/hashicorp/helm/2.9.0/docs/resources/release) | resource |
-| [helm_release.container_insights](https://registry.terraform.io/providers/hashicorp/helm/2.9.0/docs/resources/release) | resource |
```
Comment (caphene) [inline terraform/logical/README.md:37]: I'm assuming this is being replaced by adot_exporter?

**Review by caphene — APPROVED** [review]
LGTM 👍

File: terraform/logical/README.md (hunk 1 above)
Comment (ddvdozuki) [reply terraform/logical/README.md:37]: Correct

**Pull request merged**